### PR TITLE
Add GitHub release action

### LINKED ISSUE #3924: Publish prebuilt binaries for protoc-gen-go-grpc
First of all: congrats for the 1.0.0 release! 🎉 

### Use case(s) - what problem will this feature solve?

Compiling from source using `go get` does not work in every situation

### Proposed Solution

Provide prebuilt binaries for common platforms.

### Additional Context

Go protocol buffers provides builds for windows, linux, darwin and 386, amd64 platforms.

### PR DESCRIPTION
**Pull request by sagikazarmark**
This PR adds a GitHub action workflow that is triggered when a tag matching `cmd/protoc-gen-go-grpc/v[0-9]+.[0-9]+.[0-9]+` is pushed.

The workflow creates a draft release for the tag and uploads build artifacts to it as discussed in #3924. The release follows [protobuf-go](https://github.com/protocolbuffers/protobuf-go) conventions for release name assets.

An example for the created draft can be found here: https://github.com/sagikazarmark/grpc-go/releases/tag/cmd%2Fprotoc-gen-go-grpc%2Fv1.2.0
(Published, so others can see it)

Fixes #3924

### CODE REVIEW

**Comment by menghanl** [comment]
> The workflow creates a draft PR for the tag

Did you mean creates a draft _release_?

File: .github/workflows/release.yml
Comment (menghanl) [inline .github/workflows/release.yml:10]: Nit: `Create cmd/protoc-gen-go-grpc Release`

**Comment by sagikazarmark** [comment]
@menghanl yes 😄

**Review by menghanl — APPROVED** [review]
Thanks for the PR. LGTM (with my very limited github action knowledge)

File: .github/workflows/release.yml
Comment (dfawley) [inline .github/workflows/release.yml:6]: Do the dots need to be escaped?

File: .github/workflows/release.yml
Comment (dfawley) [inline .github/workflows/release.yml:46]: Is there a way to specify "use the latest stable Go release version" instead of picking a version?

**Review by dfawley — COMMENTED** [review]
This looks really cool, thank you!  A couple small comments in-line.

Is there any easy way you know of to make it so the build+upload is triggered by us creating the release from the github UI?  I'd rather avoid using the command-line to push tags if possible.  If not, that's okay -- we can definitely use this.

File: .github/workflows/release.yml
Comment (sagikazarmark) [reply .github/workflows/release.yml:6]: No, they don't. It's not actually regex.

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

File: .github/workflows/release.yml
Comment (sagikazarmark) [reply .github/workflows/release.yml:46]: Yes, without specifying the version, but I'd recommend against it. Using the latest version always carries a risk of a failing release. Updating the version twice a year shouldn't be too much effort, but I can remove it if you prefer using the latest version.

Comment (dfawley) [reply .github/workflows/release.yml]: I think I like this way (no version specified) better.  I'm not too worried about a failure here because of the Go version, and if that happens, which should be rare, we can do it manually.

**Comment by dfawley** [comment]
@sagikazarmark maybe you missed my question earlier, because it was not a separate comment?

> Is there any easy way you know of to make it so the build+upload is triggered by us creating the release from the github UI? I'd rather avoid using the command-line to push tags if possible. If not, that's okay -- we can definitely use this.

**Comment by sagikazarmark** [comment]
@dfawley yeah, I did miss your question, sorry.

There is no documented way to do it, but let me give it another try.

**Comment by sagikazarmark** [comment]
@dfawley it was easier than I thought and looks like it's working.

Creating (more precisely **publishing**) releases should work now. Publishing a release means creating a new release or pre-release, or publishing a draft release.

See example here: https://github.com/sagikazarmark/grpc-go/releases/tag/cmd%2Fprotoc-gen-go-grpc%2Fv1.4.0

**Review by dfawley — APPROVED** [review]
OK, this is super cool.  Thank you so much for doing this!

**Comment by sagikazarmark** [comment]
Thanks! Let me know if it needs anything else done.

@dfawley do you thing you could publish another release after merging to try it?

**Comment by dfawley** [comment]
> @dfawley do you thing you could publish another release after merging to try it?

Yes, we need to do another release anyway, so we will wait for this to be merged (TRAVIS?) before doing it.

**Comment by dfawley** [comment]
I performed the 1.0.1 release and the binaries were attached as expected!

On my Mac, using the downloaded tgz:

```
$ tar xzf *
$ ./protoc-gen-go-grpc -version
protoc-gen-go-grpc 1.0.1
```

Thanks again for doing this!

**Pull request merged**
